File: backend/src/main/java/com/ifeed/model/CourseEventQuestion.java
package com.ifeed.model;

import com.ifeed.model.base.IdentifiableEntity;
import com.ifeed.model.enums.QuestionState;

import javax.persistence.*;

/**
 * Created by ipascu on 19.05.2016.
 */
@Entity
@Table(name="course_event_question")
public class CourseEventQuestion extends IdentifiableEntity {

    @Column(name = "question_state", nullable = false, length = 20)
    private QuestionState questionState;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "course_event_id")
    private CourseEvent courseEvent;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "question_id")
    private Question question;

    public CourseEventQuestion() {}

    public CourseEventQuestion(QuestionState questionState, CourseEvent courseEvent, Question question) {
        this.questionState = questionState;
        this.courseEvent = courseEvent;
        this.question = question;
    }

    public QuestionState getQuestionState() {
        return questionState;
    }

    public void setQuestionState(QuestionState questionState) {
        this.questionState = questionState;
    }

    public CourseEvent getCourseEvent() {
        return courseEvent;
    }

    public void setCourseEvent(CourseEvent courseEvent) {
        this.courseEvent = courseEvent;
    }

    public Question getQuestion() {
        return question;
    }

    public void setQuestion(Question question) {
        this.question = question;
    }
}

File: frontend/src/app/data/CourseEventQuestion.ts
import Question from "./Question";
import {QuestionState} from "./enums/QuestionState";

export default class CourseEventQuestion {

    constructor(public id?: number,
                public version?: number,
                public courseEventId?: number,
                public question?: Question,
                public questionState?: QuestionState) {
    }

    static fromJson(json:any):CourseEventQuestion {
        let question: Question = Question.fromJson(json['question']);
        return new CourseEventQuestion(
            json['id'],
            json['version'],
            json['courseEventId'],
            question,
            json['questionState']);
    }

    public getProperties(): any {
        let results: Array<string> = [];

        if (this.id != null && this.version != null) {
            results.push('id=' + this.id);
            results.push('version=' + this.version);
        }
        results.push('courseEventId=' + this.courseEventId);
        results.push('questionId=' + this.question.id);
        results.push('questionState=' + this.questionState);

        return results;
    }
}